client/Api/index.js

```diff
@@ -0,0 +1,270 @@
+import axios from 'axios';
+import firebase from 'firebase';
+import { firebaseAuth, db } from '../../server/config/db';
+import PostItActions from '../actions/PostItActions';
+import PostItStore from '../stores/PostItStore';
+
+module.exports = {
+
+  /**
+   * api call to register new user from the signup route
+   *
+   * @param {any} user
+   */
+  registerNewUser(user) {
+    axios.post('/user/signup', {
+      email: user.email,
+      password: user.password,
+      userName: user.userName,
+      phoneNumber: user.phoneNumber
+    })
+    .then((response) => {
+      const authuser = {
+        email: user.email,
+        isAuthenticated: true
+      };
+      if ((response.data.message === 'The email address is badly formatted.') || (response.data.message === 'The email address is already in use by another account.')) {
+        PostItActions.receiveErrors(response.data.message);
+      } else {
+        PostItActions.receiveSuccess(response.data.message);
+        localStorage.setItem('user', response.data.user.stsTokenManager.accessToken);
+        PostItStore.setLoggedInUser(response.data.user);
+        PostItActions.receiveAuthenticatedUser(authuser);
+      }
+    })
+  .catch((error) => {
+    PostItActions.receiveErrors(error.message);
+  });
+  },
+
+  /**
+   * api call to signin user from the signin route
+   *
+   * @param {any} user
+   */
+  signinUser(user) {
+    axios.post('/user/signin', {
+      email: user.email,
+      password: user.password
+    }).then((response) => {
+      const authuser = {
+        email: user.email,
+        isAuthenticated: true
+      };
+      if (response.data.message === 'Error: The email or password of the user is invalid') {
+        PostItActions.receiveErrors(response.data.message);
+      } else {
+        PostItActions.receiveSuccess(response.data.message);
+        localStorage.setItem('user', response.data.user.stsTokenManager.accessToken);
+        PostItActions.receiveAuthenticatedUser(authuser);
+        PostItStore.setLoggedInUser(response.data.user);
+      }
+    })
+  .catch((error) => {
+    PostItActions.receiveErrors(error.message);
+  });
+  },
+
+  /**
+   * api call to signout user from the signout route
+   *
+   */
+  signoutUser() {
+    axios.get('/user/signout').then((response) => {
+      PostItActions.receiveSuccess(response.message);
+      localStorage.removeItem('user');
+    })
+    .catch((error) => {
+      PostItActions.receiveErrors(error.message);
+    });
+  },
+
+  /**
+   * api call to login us with google
+   *
+   */
+  googleLogin() {
+    let email,
+      uid,
+      displayName;
+    const provider = new firebase.auth.GoogleAuthProvider();
+    provider.addScope('https://www.googleapis.com/auth/plus.login');
+    firebase.auth().signInWithPopup(provider)
+      .then((result) => {
```

Comment:
Hi @sasili-adetunji, I feel you can make use of one .then.catch block. This will enable you better debug/catch errors where they exist. The use of promise-chains makes things a bit more complicated working with non-relational databases.